src/Altinn.Platform/Altinn.Platform.Storage/Storage/Authorization/StorageAccessHandler.cs

```diff
@@ -0,0 +1,190 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Altinn.Platform.Storage.Authorization
+{
+    /// <summary>
+    /// AuthorizationHandler that is created for handling access to app.
+    /// Authorizes based om AppAccessRequirement and app id from route
+    /// <see href="https://docs.asp.net/en/latest/security/authorization/policies.html"/> for details about authorization
+    /// in asp.net core
+    /// </summary>
+    public class StorageAccessHandler : AuthorizationHandler<AppAccessRequirement>
+    {
+        private readonly IInstanceRepository _instanceRepository;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IPDP _pdp;
+        private readonly PepSettings _pepSettings;
+        private readonly ILogger _logger;
+        private readonly IMemoryCache _memoryCache;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StorageAccessHandler"/> class.
+        /// </summary>
+        /// <param name="httpContextAccessor">The http context accessor</param>
+        /// <param name="pdp">The pdp</param>
+        /// <param name="pepSettings">The settings for pep</param>
+        /// <param name="logger">The logger. </param
+        /// <param name="instanceRepository">The instance repository</param>
+        public StorageAccessHandler(
+            IHttpContextAccessor httpContextAccessor,
+            IPDP pdp,
+            IOptions<PepSettings> pepSettings,
+            ILogger<StorageAccessHandler> logger,
+            IInstanceRepository instanceRepository,
+            IMemoryCache memoryCache)
+        {
+            _httpContextAccessor = httpContextAccessor;
+            _pdp = pdp;
+            _pepSettings = pepSettings.Value;
+            _logger = logger;
+            _instanceRepository = instanceRepository;
+            _memoryCache = memoryCache;
+        }
+
+        /// <summary>
+        /// This method authorize access bases on context and requirement
+        /// Is triggered by annotation on MVC action and setup in startup.
+        /// </summary>
+        /// <param name="context">The context</param>
+        /// <param name="requirement">The requirement</param>
+        /// <returns>A Task</returns>
+        protected override async Task HandleRequirementAsync(AuthorizationHandlerContext context, AppAccessRequirement requirement)
+        {
+            if (_pepSettings.DisablePEP)
+            {
+                context.Succeed(requirement);
+                return;
+            }
+
+            XacmlJsonRequestRoot request = DecisionHelper.CreateDecisionRequest(context, requirement, _httpContextAccessor.HttpContext.GetRouteData());
+
+            _logger.LogInformation($"// Altinn PEP // AppAccessHandler // Request sent: {JsonConvert.SerializeObject(request)}");
```

Comment:
This comment should be updated or deleted

Reply:
fixed